scarb-metadata/src/lib.rs

```diff
@@ -356,6 +380,28 @@ pub struct CompilationUnitComponentMetadata {
     /// If not specified, the one from `CompilationUnit` will be used.
     #[serde(default)]
     pub cfg: Option<Vec<Cfg>>,
+    /// Identifier of this component. It is unique in its compilation unit.
+    pub id: Option<CompilationUnitComponentId>,
+    /// Dependencies of this component.
+    pub dependencies: Option<Vec<CompilationUnitComponentDependencyMetadata>>,
```

Comment:
Do we need to differentiate between `Option::None` and empty `Vec`?

Reply:
Yes, because crate with no dependencies (empty vec) is a valid thing. And scarb metadata from old days (None) is also a valid thing and in such cache all components should be assumed as dependencies